src/formatDate.js

```diff
@@ -8,7 +8,61 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.find(
+    (item) => item === '/' || item === '-' || item === '.',
+  );
+
+  const separator = toFormat.find(
+    (item) => item === '/' || item === '-' || item === '.',
+  );
+
+  const newDate = date.split(fromSeparator);
+
+  const newArray = [];
+
+  const dateObject = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = newDate[i];
+  }
```

Comment:
The loop iterates `fromFormat.length - 1` times, which might cause issues if the last element in `fromFormat` is needed. Consider iterating through the entire `fromFormat` array.